[docs] Update v1 docs to mention google-services.json (#27658)
expo/expo: docs/pages/push-notifications/fcm-credentials.mdx

description: Learn how to create or use a Google Service Account Key for sending Android Notifications using FCM V1.
---

> import { CodeBlocksTable } from '~/components/plugins/CodeBlocksTable';

Reviewer: ```suggestion
```

We don't need this. Please see my comment below.

</Step>

<Step label="5">

>     Configure the **google-services.json** file in your project. Download it from the Firebase Console and place it at the root of your project directory. If you're using version control, be sure to add this to your ignore file (e.g. **.gitignore**) as it contains sensitive data.

Reviewer: ```suggestion
    Configure the **google-services.json** file in your project. Download it from the Firebase Console and place it at the root of your project directory. If you're using version control, add it to your ignore file (for example, **.gitignore**) as it contains sensitive data.
```

- Suggestion to update the verbiage slightly for improved readability
- Also, let's avoid [Latin abbreviations](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#abbreviations) as per our writing style guide.

</Step>

<Step label="5">

>   Configure the `google-services.json` file in your app. (Note: you may skip this step if `google-services.json` has already been set up.)
> 
>   First, download the `google-services.json` file from Firebase Cloud Console and place it somewhere in your app directory (the root is fine). 

Reviewer: ```suggestion
  Configure the **google-services.json** file in your project. Download it from the Firebase Console and place it at the root of your project directory.
  
  > **Note**: You can skip this step if **google-services.json** has already been set up.
```

- Let's make [file names bold](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#filenames-as-bold-text) as per our writing style guide
- In this guide, above, we're using Firebase Console instead of Firebase Cloud Console. Either we use the former here too or change it above as per this change (to use the latter).
- We should be authoritative about placing the file so I think informing about placing it at the root for the project is fine in this case. We can also be explicit about it by adding the context, placing it anywhere else in the project directory is also fine.
- Let's move the not out from the first line and make it a callout.

  <ImageSpotlight
    alt="Download google-services.json from Firebase Cloud Console"
    src="/static/images/creating-google-service-account/fcm-v1/new-service-account/05-setup-google-services-json.png"
    style={{ maxWidth: 800 }}
  />

>   Then update your app.json with the key `expo.android.googleServicesFile` and value `path/to/google-services.json`.

Reviewer: ```suggestion
  Then,  in **app.json**, add [`expo.android.googleServicesFile`](/versions/latest/config/app/#googleservicesfile) with its value as the path of the **google-services.json**.
```

- Let's add an internal link to the `android.googleServicesFile` and also mention in words about its value instead of using inline code to highlight that.

A question about this, should we also mention that in case if someone is using Git or any other Version Control System and have public reprository, then, they should add this file to their project's ** .gitignore**? Since this file contains secret values.

Author: Good call on the .gitignore, I think that's worth mentioning.

>   <CodeBlocksTable>
>     ```json
> 	// app.json
> 	{
> 	  "expo": {
> 		...
> 		"android": {
> 		  ...
> 		  "googleServicesFile": "./path/to/google-services.json"
> 		},
> 	  }
> 	}
>     ```
>   </CodeBlocksTable>

Reviewer: ````suggestion
    ```json app.json

    {
      "expo": {
    	/* @hide ...*/ /* @end */
    	"android": {
    	  /* @hide ...*/ /* @end */
    	  "googleServicesFile": "./path/to/google-services.json"
    	},
      }
    }
    ```
````

We don't have to use CodeBlocksTable here unless there is a side-by-side comparison of two code snippets  Currently, this is how it renders:
![CleanShot 2024-03-14 at 17 58 38](https://github.com/expo/expo/assets/10234615/1f521393-9337-4656-a7f6-3bf4d36127ee)

- Let's remove the component and use the code snippet display pattern as we usually do in our docs (such as instead of using three dots, we use `@hide` annotation and the file name can be with the snippet language code)

</Step>

<Step label="4">

>     Configure the **google-services.json** file in your project. Download it from the Firebase Console and place it at the root of your project directory. If you're using version control, be sure to add this to your ignore file (e.g. **.gitignore**) as it contains sensitive data.

Reviewer: ```suggestion
    Configure the **google-services.json** file in your project. Download it from the Firebase Console and place it at the root of your project directory. If you're using version control, add it to your ignore file (for example, **.gitignore**) as it contains sensitive data.
```

Same as above

</Step>

<Step label="4">

>   Configure the `google-services.json` file in your app. (Note: you may skip this step if `google-services.json` has already been set up.)
> 
>   First, download the `google-services.json` file from Firebase Cloud Console and place it somewhere in your app directory (the root is fine). 

Reviewer: ```suggestion
  Configure the **google-services.json** file in your project. Download it from the Firebase Console and place it at the root of your project directory.
  
  > **Note**: You can skip this step if **google-services.json** has already been set up.
```

Same as above.

  <ImageSpotlight
    alt="Download google-services.json from Firebase Cloud Console"
    src="/static/images/creating-google-service-account/fcm-v1/existing-service-account/04-setup-google-services-json.png"
    style={{ maxWidth: 800 }}
  />

  Then update your app.json with the key `expo.android.googleServicesFile` and value `path/to/google-services.json`.

>   <CodeBlocksTable>
>     ```json
> 	// app.json
> 	{
> 	  "expo": {
> 		...
> 		"android": {
> 		  ...
> 		  "googleServicesFile": "./path/to/google-services.json"
> 		},
> 	  }
> 	}
>     ```
>   </CodeBlocksTable>

Reviewer: ````suggestion
    ```json app.json
    {
      "expo": {
    	/* @hide ...*/ /* @end */
    	"android": {
    	  /* @hide ...*/ /* @end */
    	  "googleServicesFile": "./path/to/google-services.json"
    	},
      }
    }
    ```
````
